std/common.sy

```diff
@@ -1,22 +1,22 @@
-dbg :: fn x ->
+dbg :: pu x ->
     print(x)
     x
 end
 
-args: fn -> {str:str} : external
+args: pu -> {str:str} : external
 thread_sleep: fn float -> void : external
-print: fn *X -> void : external
+print: pu *X -> void : external
```

Comment:
hmmm

Reply:
Oops, hehe

Reply:
I should probably look over this some more, I realize now I just put pure everywhere without a second thought.

Reply:
honestly it depends on what we consider "pure". is it only pure with respect to variables or is external code allowed to do stuff with the system as long as it isn't visible to sylt-code? one could consider "pure" as something like "const" from rust.

Reply:
I like the idea of not having side-effects at all. So all operations that manipulate a list or change the state or output to the console are considered impure. But we should probably allow `spy` and `dbg` too make debugging eaiser. :D

Reply:
haskell has https://hackage.haskell.org/package/base-4.16.0.0/docs/Debug-Trace.html for specifically that reason